src/main/java/io/snowdrop/github/reporting/IssueCollector.java

```diff
@@ -36,30 +37,33 @@
   private final Set<String> organizations;
 
   // These dates represent current reporting period
-  private final ZonedDateTime startTime;
-  private final ZonedDateTime endTime;
+  private ZonedDateTime startTime;
+  private ZonedDateTime endTime;
 
   // These represent the oldest reporting period possible
-  private final Date minStartTime;
-  private final Date minEndTime;
+  private Date minStartTime;
+  private Date minEndTime;
 
-  public IssueCollector(GitHubClient client, StatusLogger status, int reportingDay, int reportingHour, Set<String> users,
+  public IssueCollector(
+      GitHubClient client, StatusLogger status, int reportingDay, int reportingHour, Set<String> users,
       Set<String> organizations) {
     this.client = client;
     this.status = status;
     this.issueService = new IssueService(client);
     this.users = users;
     this.organizations = organizations;
+    init(Optional.of(reportingDay), Optional.of(reportingHour));
+  }
 
-    this.endTime = ZonedDateTime.now().with(DayOfWeek.of(reportingDay)).withHour(reportingHour);
+  public void init(final Optional<Integer> reportingDay, final Optional<Integer> reportingHour) {
```

Comment:
The name init implies that it should only be called during intialization, however it's called each time we collect in order to refresh the endTime. A more fitting name is needed here.

Reply:
I'll add a better name.